android/src/main/java/com/mycompany/plugins/example/AppIconBase.java

```diff
@@ -35,9 +39,11 @@ public Boolean isSupported() {
     }
 
     public String getName() {
-        ComponentName componentName = new ComponentName(this.activity, this.activity.getClass());
+        Intent intent = pm.getLaunchIntentForPackage(context.getPackageName());
+        ComponentName componentName = intent.getComponent();
         int status = pm.getComponentEnabledSetting(componentName);
-        if (status == PackageManager.COMPONENT_ENABLED_STATE_ENABLED) {
+
+        if (status == PackageManager.COMPONENT_ENABLED_STATE_ENABLED || status == PackageManager.COMPONENT_ENABLED_STATE_DEFAULT) {
```

Comment:
I'm not really sure if this `if()` statement is needed, it's not used in the https://github.com/true-hamid/rn-dynamic-app-icon library.